crates/belly_core/src/relations/mod.rs

```diff
@@ -200,20 +184,19 @@ impl BindingSystemsInternal {
             TypeId::of::<S>(),
             TypeId::of::<T>(),
         );
-        if self.watchers.read().unwrap().contains(&watcher) {
-            return;
-        }
-        {
-            let watchers = self.watchers.write().unwrap();
-            if watchers.contains(&watcher) {
-                return;
+        if !self.watchers.read().unwrap().contains(&watcher) {
```

Comment:
If two watches would contain two watchers, which watch would watch which watcher

Reply:
previous version with early returns was nicer. but yeah, does not work the same.